tar_scm.py

```diff
@@ -72,6 +72,17 @@ def fetch_upstream_git(url, clone_dir, revision, cwd, kwargs):
     """Fetch sources via git."""
     safe_run(['git', 'clone', url, clone_dir], cwd=cwd,
              interactive=sys.stdout.isatty())
+    if revision:
+        # check if the reference already exists.
+        try:
+            safe_run(['git', 'rev-parse', '--verify', '--quiet', revision],
```

Comment:
Haha, good catch :-)  Actually I didn't write that code.  But well done for copying the existing style, even if I don't like it ;-)  Yeah I think it would be better not to use `safe_run` when the command can fail in normal circumstances.

Also I now notice that `switch_revision_git` currently needs to check if a revision exists.  If we're adding a check here, then maybe that's no longer necessary - what do you think?